Handle librsvg deprecations

Building against `librsvg 2.52.0`, the following deprecation warnings are emitted:

```
/home/sakari/builds/darktable-git/src/darktable/src/common/utility.c: In function ‘_util_get_svg_img’:
/home/sakari/builds/darktable-git/src/darktable/src/common/utility.c:422:5: warning: ‘rsvg_handle_get_dimensions’ is deprecated: Use 'rsvg_handle_get_intrinsic_size_in_pixels' instead [-Wdeprecated-declarations]
  422 |     rsvg_handle_get_dimensions(svg, &dimension);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/sakari/builds/darktable-git/src/darktable/src/common/utility.c:52:
/usr/include/librsvg-2.0/librsvg/rsvg.h:719:6: note: declared here
  719 | void rsvg_handle_get_dimensions (RsvgHandle *handle, RsvgDimensionData *dimension_data);
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sakari/builds/darktable-git/src/darktable/src/common/utility.c:451:7: warning: ‘rsvg_handle_render_cairo’ is deprecated: Use 'rsvg_handle_render_document' instead [-Wdeprecated-declarations]
  451 |       rsvg_handle_render_cairo(svg, cr);
      |       ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/librsvg-2.0/librsvg/rsvg.h:1452,
                 from /home/sakari/builds/darktable-git/src/darktable/src/common/utility.c:52:
/usr/include/librsvg-2.0/librsvg/rsvg-cairo.h:95:10: note: declared here
   95 | gboolean rsvg_handle_render_cairo (RsvgHandle *handle, cairo_t *cr);
```

In fact these are turned into errors with the default build flags, so current master is unbuildable against `librsvg 2.52.0`.

--- a/src/common/utility.c
+++ b/src/common/utility.c
@@ -419,7 +419,7 @@ static cairo_surface_t *_util_get_svg_img(gchar *logo, const float size)
   if(svg)
   {
     RsvgDimensionData dimension;
-    dimension = getSvgDimension(*svg);
+    dimension = getSvgDimension(svg);
 
     const float ppd = darktable.gui ? darktable.gui->ppd : 1.0;
 
@@ -448,7 +448,7 @@ static cairo_surface_t *_util_get_svg_img(gchar *logo, const float size)
     {
       cairo_t *cr = cairo_create(surface);
       cairo_scale(cr, factor, factor);      
-      dt_render_svg(*svg, cr, final_width, final_height, 0, 0);
+      dt_render_svg(svg, cr, final_width, final_height, 0, 0);
       cairo_destroy(cr);
       cairo_surface_flush(surface);
     }
@@ -889,14 +889,14 @@ void dt_copy_resource_file(const char *src, const char *dst)
   g_free(sourcefile);
 }
 
-RsvgDimensionData getSvgDimension(RsvgHandle svg) 
+RsvgDimensionData getSvgDimension(RsvgHandle *svg) 
 {
   RsvgDimensionData dimension;
   // rsvg_handle_get_dimensions has been deprecated in librsvg 2.52
   #if LIBRSVG_CHECK_VERSION(2,52,0)
     double width;
     double height;
-    rsvg_handle_get_intrinsic_size_in_pixels(&svg, &width, &height);
+    rsvg_handle_get_intrinsic_size_in_pixels(svg, &width, &height);
     dimension.width = width;
     dimension.height = height;
   #else      
@@ -905,7 +905,7 @@ RsvgDimensionData getSvgDimension(RsvgHandle svg)
   return dimension; 
 }
 
-void dt_render_svg(RsvgHandle svg, cairo_t *cr, double width, double height, double offset_x, double offset_y) 
+void dt_render_svg(RsvgHandle *svg, cairo_t *cr, double width, double height, double offset_x, double offset_y) 
 {
   // rsvg_handle_render_cairo has been deprecated in librsvg 2.52
   #if LIBRSVG_CHECK_VERSION(2,52,0)
@@ -915,9 +915,9 @@ void dt_render_svg(RsvgHandle svg, cairo_t *cr, double width, double height, dou
       .width = width,
       .height = height,
     };
-    rsvg_handle_render_document(&svg, cr, &viewport, NULL);
+    rsvg_handle_render_document(svg, cr, &viewport, NULL);
   #else
-    rsvg_handle_render_cairo(&svg, cr);
+    rsvg_handle_render_cairo(svg, cr);
   #endif  
 }
 

--- a/src/common/utility.c
+++ b/src/common/utility.c
@@ -417,9 +417,18 @@ static cairo_surface_t *_util_get_svg_img(gchar *logo, const float size)
   char *dtlogo = g_build_filename(datadir, "pixmaps", logo, NULL);
   RsvgHandle *svg = rsvg_handle_new_from_file(dtlogo, &error);
   if(svg)
-  {
+  {  
     RsvgDimensionData dimension;
-    rsvg_handle_get_dimensions(svg, &dimension);
+    // rsvg_handle_get_dimensions has been deprecated in librsvg 2.52
+    #if LIBRSVG_CHECK_VERSION(2,52,0)
+      double width;
+      double height;
+      rsvg_handle_get_intrinsic_size_in_pixels(svg, &width, &height);
+      dimension.width = width;
+      dimension.height = height;
+    #else      
+      rsvg_handle_get_dimensions(svg, &dimension);
+    #endif 
 
     const float ppd = darktable.gui ? darktable.gui->ppd : 1.0;
 
@@ -448,7 +457,19 @@ static cairo_surface_t *_util_get_svg_img(gchar *logo, const float size)
     {
       cairo_t *cr = cairo_create(surface);
       cairo_scale(cr, factor, factor);
-      rsvg_handle_render_cairo(svg, cr);
+      
+      // rsvg_handle_render_cairo has been deprecated in librsvg 2.52
+      #if LIBRSVG_CHECK_VERSION(2,52,0)
+        RsvgRectangle viewport;
+	      viewport.x = 0;
+	      viewport.y = 0;
+	      viewport.width = final_width;
+	      viewport.height = final_height;
+        rsvg_handle_render_document(svg, cr, &viewport, &error);
+      #else
+        rsvg_handle_render_cairo(svg, cr);
+      #endif  
+
       cairo_destroy(cr);
       cairo_surface_flush(surface);
     }

--- a/src/common/utility.c
+++ b/src/common/utility.c
@@ -417,7 +417,7 @@ static cairo_surface_t *_util_get_svg_img(gchar *logo, const float size)
   char *dtlogo = g_build_filename(datadir, "pixmaps", logo, NULL);
   RsvgHandle *svg = rsvg_handle_new_from_file(dtlogo, &error);
   if(svg)
-  {  
+  {
     RsvgDimensionData dimension;
     // rsvg_handle_get_dimensions has been deprecated in librsvg 2.52
     #if LIBRSVG_CHECK_VERSION(2,52,0)

--- a/src/common/utility.c
+++ b/src/common/utility.c
@@ -419,16 +419,7 @@ static cairo_surface_t *_util_get_svg_img(gchar *logo, const float size)
   if(svg)
   {
     RsvgDimensionData dimension;
-    // rsvg_handle_get_dimensions has been deprecated in librsvg 2.52
-    #if LIBRSVG_CHECK_VERSION(2,52,0)
-      double width;
-      double height;
-      rsvg_handle_get_intrinsic_size_in_pixels(svg, &width, &height);
-      dimension.width = width;
-      dimension.height = height;
-    #else      
-      rsvg_handle_get_dimensions(svg, &dimension);
-    #endif 
+    dimension = getSvgDimension(*svg);
 
     const float ppd = darktable.gui ? darktable.gui->ppd : 1.0;
 
@@ -456,20 +447,8 @@ static cairo_surface_t *_util_get_svg_img(gchar *logo, const float size)
     else
     {
       cairo_t *cr = cairo_create(surface);
-      cairo_scale(cr, factor, factor);
-      
-      // rsvg_handle_render_cairo has been deprecated in librsvg 2.52
-      #if LIBRSVG_CHECK_VERSION(2,52,0)
-        RsvgRectangle viewport;
-	      viewport.x = 0;
-	      viewport.y = 0;
-	      viewport.width = final_width;
-	      viewport.height = final_height;
-        rsvg_handle_render_document(svg, cr, &viewport, &error);
-      #else
-        rsvg_handle_render_cairo(svg, cr);
-      #endif  
-
+      cairo_scale(cr, factor, factor);      
+      dt_render_svg(*svg, cr, final_width, final_height, 0, 0);
       cairo_destroy(cr);
       cairo_surface_flush(surface);
     }
@@ -908,6 +887,38 @@ void dt_copy_resource_file(const char *src, const char *dst)
   gchar *sourcefile = g_build_filename(share, src, NULL);
   dt_copy_file(sourcefile, dst);
   g_free(sourcefile);
+}
+
+RsvgDimensionData getSvgDimension(RsvgHandle svg) 
+{
+  RsvgDimensionData dimension;
+  // rsvg_handle_get_dimensions has been deprecated in librsvg 2.52
+  #if LIBRSVG_CHECK_VERSION(2,52,0)
+    double width;
+    double height;
+    rsvg_handle_get_intrinsic_size_in_pixels(&svg, &width, &height);
+    dimension.width = width;
+    dimension.height = height;
+  #else      
+    rsvg_handle_get_dimensions(&svg, &dimension);
+  #endif
+  return dimension; 
+}
+
+void dt_render_svg(RsvgHandle svg, cairo_t *cr, double width, double height, double offset_x, double offset_y) 
+{
+  // rsvg_handle_render_cairo has been deprecated in librsvg 2.52
+  #if LIBRSVG_CHECK_VERSION(2,52,0)
+    RsvgRectangle viewport = { 
+      .x = offset_x,
+      .y = offset_y,
+      .width = width,
+      .height = height,
+    };
+    rsvg_handle_render_document(&svg, cr, &viewport, NULL);
+  #else
+    rsvg_handle_render_cairo(&svg, cr);
+  #endif  
 }
 
 // modelines: These editor modelines have been set for all relevant files by tools/update_modelines.sh

--- a/src/common/utility.c
+++ b/src/common/utility.c
@@ -420,7 +420,7 @@ static cairo_surface_t *_util_get_svg_img(gchar *logo, const float size)
   {
     RsvgDimensionData dimension;
     // rsvg_handle_get_dimensions has been deprecated in librsvg 2.52
-    #if LIBRSVG_CHECK_VERSION(2,46,0)
+    #if LIBRSVG_CHECK_VERSION(2,52,0)
       double width;
       double height;
       rsvg_handle_get_intrinsic_size_in_pixels(svg, &width, &height);
@@ -459,7 +459,7 @@ static cairo_surface_t *_util_get_svg_img(gchar *logo, const float size)
       cairo_scale(cr, factor, factor);
       
       // rsvg_handle_render_cairo has been deprecated in librsvg 2.52
-      #if LIBRSVG_CHECK_VERSION(2,46,0)
+      #if LIBRSVG_CHECK_VERSION(2,52,0)
         RsvgRectangle viewport;
 	      viewport.x = 0;
 	      viewport.y = 0;

--- a/src/common/utility.c
+++ b/src/common/utility.c
@@ -419,7 +419,7 @@ static cairo_surface_t *_util_get_svg_img(gchar *logo, const float size)
   if(svg)
   {
     RsvgDimensionData dimension;
-    dimension = getSvgDimension(svg);
+    dimension = dt_get_svg_dimension(svg);
 
     const float ppd = darktable.gui ? darktable.gui->ppd : 1.0;
 
@@ -889,7 +889,7 @@ void dt_copy_resource_file(const char *src, const char *dst)
   g_free(sourcefile);
 }
 
-RsvgDimensionData getSvgDimension(RsvgHandle *svg) 
+RsvgDimensionData dt_get_svg_dimension(RsvgHandle *svg) 
 {
   RsvgDimensionData dimension;
   // rsvg_handle_get_dimensions has been deprecated in librsvg 2.52

--- a/src/common/utility.c
+++ b/src/common/utility.c
@@ -447,7 +447,7 @@ static cairo_surface_t *_util_get_svg_img(gchar *logo, const float size)
     else
     {
       cairo_t *cr = cairo_create(surface);
-      cairo_scale(cr, factor, factor);      
+      cairo_scale(cr, factor, factor);
       dt_render_svg(svg, cr, final_width, final_height, 0, 0);
       cairo_destroy(cr);
       cairo_surface_flush(surface);

--- a/src/common/utility.c
+++ b/src/common/utility.c
@@ -900,7 +900,7 @@ RsvgDimensionData getSvgDimension(RsvgHandle *svg)
     dimension.width = width;
     dimension.height = height;
   #else      
-    rsvg_handle_get_dimensions(&svg, &dimension);
+    rsvg_handle_get_dimensions(svg, &dimension);
   #endif
   return dimension; 
 }

--- a/src/common/utility.c
+++ b/src/common/utility.c
@@ -420,7 +420,7 @@ static cairo_surface_t *_util_get_svg_img(gchar *logo, const float size)
   {
     RsvgDimensionData dimension;
     // rsvg_handle_get_dimensions has been deprecated in librsvg 2.52
-    #if LIBRSVG_CHECK_VERSION(2,52,0)
+    #if LIBRSVG_CHECK_VERSION(2,46,0)
       double width;
       double height;
       rsvg_handle_get_intrinsic_size_in_pixels(svg, &width, &height);
@@ -459,7 +459,7 @@ static cairo_surface_t *_util_get_svg_img(gchar *logo, const float size)
       cairo_scale(cr, factor, factor);
       
       // rsvg_handle_render_cairo has been deprecated in librsvg 2.52
-      #if LIBRSVG_CHECK_VERSION(2,52,0)
+      #if LIBRSVG_CHECK_VERSION(2,46,0)
         RsvgRectangle viewport;
 	      viewport.x = 0;
 	      viewport.y = 0;

--- a/src/common/utility.h
+++ b/src/common/utility.h
@@ -99,10 +99,10 @@ void dt_copy_file(const char *src, const char *dst);
 void dt_copy_resource_file(const char *src, const char *dst);
 
 // returns the RsvgDimensionData of a supplied RsvgHandle
-RsvgDimensionData getSvgDimension(RsvgHandle svg);
+RsvgDimensionData getSvgDimension(RsvgHandle *svg);
 
 // renders svg data
-void dt_render_svg(RsvgHandle svg, cairo_t *cr, double width, double height, double offset_x, double offset_y);
+void dt_render_svg(RsvgHandle *svg, cairo_t *cr, double width, double height, double offset_x, double offset_y);
 
 // modelines: These editor modelines have been set for all relevant files by tools/update_modelines.sh
 // vim: shiftwidth=2 expandtab tabstop=2 cindent

--- a/src/common/utility.h
+++ b/src/common/utility.h
@@ -20,6 +20,7 @@
 
 #include <gtk/gtk.h>
 #include <string.h>
+#include <librsvg/rsvg.h>
 
 /** dynamically allocate and concatenate string */
 gchar *dt_util_dstrcat(gchar *str, const gchar *format, ...) __attribute__((format(printf, 2, 3)));
@@ -96,6 +97,12 @@ void dt_copy_file(const char *src, const char *dst);
 
 // copy the contents of a file in dt's data directory to a new file
 void dt_copy_resource_file(const char *src, const char *dst);
+
+// returns the RsvgDimensionData of a supplied RsvgHandle
+RsvgDimensionData getSvgDimension(RsvgHandle svg);
+
+// renders svg data
+void dt_render_svg(RsvgHandle svg, cairo_t *cr, double width, double height, double offset_x, double offset_y);
 
 // modelines: These editor modelines have been set for all relevant files by tools/update_modelines.sh
 // vim: shiftwidth=2 expandtab tabstop=2 cindent

--- a/src/common/utility.h
+++ b/src/common/utility.h
@@ -99,7 +99,7 @@ void dt_copy_file(const char *src, const char *dst);
 void dt_copy_resource_file(const char *src, const char *dst);
 
 // returns the RsvgDimensionData of a supplied RsvgHandle
-RsvgDimensionData getSvgDimension(RsvgHandle *svg);
+RsvgDimensionData dt_get_svg_dimension(RsvgHandle *svg);
 
 // renders svg data
 void dt_render_svg(RsvgHandle *svg, cairo_t *cr, double width, double height, double offset_x, double offset_y);

--- a/src/iop/watermark.c
+++ b/src/iop/watermark.c
@@ -727,7 +727,7 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const 
   {
     case DT_WTM_SVG:      
       // rsvg_handle_get_dimensions has been deprecated in librsvg 2.52
-      #if LIBRSVG_CHECK_VERSION(2,46,0)
+      #if LIBRSVG_CHECK_VERSION(2,52,0)
         double width;
         double height;
         rsvg_handle_get_intrinsic_size_in_pixels(svg, &width, &height);
@@ -923,7 +923,7 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const 
       /* render svg into surface*/
       
       // rsvg_handle_render_cairo has been deprecated in librsvg 2.52
-      #if LIBRSVG_CHECK_VERSION(2,46,0)
+      #if LIBRSVG_CHECK_VERSION(2,52,0)
         GError *error = NULL;
         RsvgRectangle viewport;
 	      viewport.x = 0;

--- a/src/iop/watermark.c
+++ b/src/iop/watermark.c
@@ -725,8 +725,7 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const 
   RsvgDimensionData dimension;
   switch(type)
   {
-    case DT_WTM_SVG:
-      
+    case DT_WTM_SVG:      
       // rsvg_handle_get_dimensions has been deprecated in librsvg 2.52
       #if LIBRSVG_CHECK_VERSION(2,52,0)
         double width;
@@ -737,8 +736,7 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const 
       #else
         rsvg_handle_get_dimensions(svg, &dimension);
       #endif 
-      break;
-      
+      break;      
     case DT_WTM_PNG:
       // load png into surface 2
       surface_two = cairo_image_surface_create_from_png(filename);

--- a/src/iop/watermark.c
+++ b/src/iop/watermark.c
@@ -726,7 +726,7 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const 
   switch(type)
   {
     case DT_WTM_SVG:      
-      dimension = getSvgDimension(*svg);
+      dimension = getSvgDimension(svg);
       break;      
     case DT_WTM_PNG:
       // load png into surface 2
@@ -912,7 +912,7 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const 
     case DT_WTM_SVG:
       cairo_scale(cr_two, scale, scale);
       /* render svg into surface*/
-      dt_render_svg(*svg, cr_two, dimension.width, dimension.height, 0, 0);
+      dt_render_svg(svg, cr_two, dimension.width, dimension.height, 0, 0);
       break;
     case DT_WTM_PNG:
       cairo_scale(cr, scale, scale);

--- a/src/iop/watermark.c
+++ b/src/iop/watermark.c
@@ -725,9 +725,9 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const 
   RsvgDimensionData dimension;
   switch(type)
   {
-    case DT_WTM_SVG:      
+    case DT_WTM_SVG:
       dimension = getSvgDimension(svg);
-      break;      
+      break;
     case DT_WTM_PNG:
       // load png into surface 2
       surface_two = cairo_image_surface_create_from_png(filename);

--- a/src/iop/watermark.c
+++ b/src/iop/watermark.c
@@ -726,7 +726,7 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const 
   switch(type)
   {
     case DT_WTM_SVG:
-      dimension = getSvgDimension(svg);
+      dimension = dt_get_svg_dimension(svg);
       break;
     case DT_WTM_PNG:
       // load png into surface 2

--- a/src/iop/watermark.c
+++ b/src/iop/watermark.c
@@ -727,7 +727,7 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const 
   {
     case DT_WTM_SVG:      
       // rsvg_handle_get_dimensions has been deprecated in librsvg 2.52
-      #if LIBRSVG_CHECK_VERSION(2,52,0)
+      #if LIBRSVG_CHECK_VERSION(2,46,0)
         double width;
         double height;
         rsvg_handle_get_intrinsic_size_in_pixels(svg, &width, &height);
@@ -923,7 +923,7 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const 
       /* render svg into surface*/
       
       // rsvg_handle_render_cairo has been deprecated in librsvg 2.52
-      #if LIBRSVG_CHECK_VERSION(2,52,0)
+      #if LIBRSVG_CHECK_VERSION(2,46,0)
         GError *error = NULL;
         RsvgRectangle viewport;
 	      viewport.x = 0;

--- a/src/iop/watermark.c
+++ b/src/iop/watermark.c
@@ -726,16 +726,7 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const 
   switch(type)
   {
     case DT_WTM_SVG:      
-      // rsvg_handle_get_dimensions has been deprecated in librsvg 2.52
-      #if LIBRSVG_CHECK_VERSION(2,52,0)
-        double width;
-        double height;
-        rsvg_handle_get_intrinsic_size_in_pixels(svg, &width, &height);
-        dimension.width = width;
-        dimension.height = height;
-      #else
-        rsvg_handle_get_dimensions(svg, &dimension);
-      #endif 
+      dimension = getSvgDimension(*svg);
       break;      
     case DT_WTM_PNG:
       // load png into surface 2
@@ -921,19 +912,7 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const 
     case DT_WTM_SVG:
       cairo_scale(cr_two, scale, scale);
       /* render svg into surface*/
-      
-      // rsvg_handle_render_cairo has been deprecated in librsvg 2.52
-      #if LIBRSVG_CHECK_VERSION(2,52,0)
-        GError *error = NULL;
-        RsvgRectangle viewport;
-	      viewport.x = 0;
-	      viewport.y = 0;
-	      viewport.width = dimension.width;
-	      viewport.height = dimension.height;
-        rsvg_handle_render_document(svg, cr_two, &viewport, &error);
-      #else
-        rsvg_handle_render_cairo(svg, cr_two);
-      #endif  
+      dt_render_svg(*svg, cr_two, dimension.width, dimension.height, 0, 0);
       break;
     case DT_WTM_PNG:
       cairo_scale(cr, scale, scale);

--- a/src/iop/watermark.c
+++ b/src/iop/watermark.c
@@ -726,8 +726,19 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const 
   switch(type)
   {
     case DT_WTM_SVG:
-      rsvg_handle_get_dimensions(svg, &dimension);
+      
+      // rsvg_handle_get_dimensions has been deprecated in librsvg 2.52
+      #if LIBRSVG_CHECK_VERSION(2,52,0)
+        double width;
+        double height;
+        rsvg_handle_get_intrinsic_size_in_pixels(svg, &width, &height);
+        dimension.width = width;
+        dimension.height = height;
+      #else
+        rsvg_handle_get_dimensions(svg, &dimension);
+      #endif 
       break;
+      
     case DT_WTM_PNG:
       // load png into surface 2
       surface_two = cairo_image_surface_create_from_png(filename);
@@ -912,7 +923,19 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const 
     case DT_WTM_SVG:
       cairo_scale(cr_two, scale, scale);
       /* render svg into surface*/
-      rsvg_handle_render_cairo(svg, cr_two);
+      
+      // rsvg_handle_render_cairo has been deprecated in librsvg 2.52
+      #if LIBRSVG_CHECK_VERSION(2,52,0)
+        GError *error = NULL;
+        RsvgRectangle viewport;
+	      viewport.x = 0;
+	      viewport.y = 0;
+	      viewport.width = dimension.width;
+	      viewport.height = dimension.height;
+        rsvg_handle_render_document(svg, cr_two, &viewport, &error);
+      #else
+        rsvg_handle_render_cairo(svg, cr_two);
+      #endif  
       break;
     case DT_WTM_PNG:
       cairo_scale(cr, scale, scale);

